src/main/java/org/terasology/taskSystem/AssignedTaskType.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.taskSystem;
+
+public class AssignedTaskType {
```

Comment:
AssignedTaskType should be an Enum, not a list of strings.
It stores better, it transmits over the network better, it compares better.  It can be used directly in switch statements. It typechecks.
The only place it falls short is that it is not extendable without changing the code.
But if that becomes an issue, even then, it could be replaced by a different extendable-enum-like class that was.

Reply:
I tried the `Enum` approach initially but the conditional BT node does not support that, so for the check to work I had to use Strings.